azure-pipelines.yml

```diff
@@ -57,6 +57,114 @@ stages:
+                  else
+                      FINAL_FEATURES="$FINAL_FEATURES,${FEATURE#tests/}"
+                  fi
+              fi
+          done
+
+          if [ -z "$FINAL_FEATURES" ]; then
+              TEST_SCRIPTS=$(python ./.azure-pipelines/validate_diff_folders/get_test_scripts.py --features "" --location tests)
+          else
+              TEST_SCRIPTS=$(python ./.azure-pipelines/validate_diff_folders/get_test_scripts.py --features ${FINAL_FEATURES} --location tests)
+          fi
+
+          PR_CHECKERS=$(echo "${TEST_SCRIPTS}" | jq -c 'keys')
+
+          echo "##vso[task.setvariable variable=PR_CHECKERS;isOutput=true]$PR_CHECKERS"
+          echo "##vso[task.setvariable variable=TEST_SCRIPTS;isOutput=true]$TEST_SCRIPTS"
+        name: SetVariableTask
+        displayName: "Get impacted area"
+
+  - job: t0_elastictest_on_needed
+    displayName: "kvmtest-t0-on-needed by Elastictest"
+    dependsOn: get_impacted_area
+    condition: or(contains(dependencies.get_impacted_area.outputs['SetVariableTask.PR_CHECKERS'], 't0'), eq(dependencies.get_impacted_area.outputs['SetVariableTask.PR_CHECKERS'], ''))
+    variables:
+      TEST_SCRIPTS: $[ dependencies.get_impacted_area.outputs['SetVariableTask.TEST_SCRIPTS'] ]
+    timeoutInMinutes: 240
+    continueOnError: true
+    pool: sonic-ubuntu-1c
+    steps:
+      - script: |
+          # Check if azure cli is installed. If not, try to install it
+          if ! command -v az; then
+            echo "Azure CLI is not installed. Trying to install it..."
+            curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+          else
+            echo "Azure CLI is already installed"
+          fi
+
+          sudo apt-get update && sudo apt-get install -y jq
+        displayName: "Install azure-cli"
+
+      - script: |
+          set -x
+
+          sudo apt-get update && sudo apt-get install -y jq
+
+          T0_SCRIPTS=$(echo '$(TEST_SCRIPTS)' | jq -r -c '.t0')
+          SCRIPTS=$(echo "$T0_SCRIPTS" | jq -r '. | join(",")')
+          echo -n "##vso[task.setvariable variable=SCRIPTS]$SCRIPTS"
+        displayName: "Get t0 test scripts"
+
+      - task: AzureCLI@2
+        displayName: "Calculate instance number"
+        inputs:
+          azureSubscription: "SONiC-Automation"
+          scriptType: 'bash'
+          scriptLocation: 'inlineScript'
+          inlineScript: |
+            set -x
+
+            pip install azure-kusto-data
+            pip install azure-kusto-data azure-identity
+
+            accessToken=$(az account get-access-token --resource https://api.kusto.windows.net --query accessToken -o tsv)
+            export ACCESS_TOKEN=$accessToken
+
+            INSTANCE_NUMBER=$(python ./.azure-pipelines/validate_diff_folders/calculate_instance_number.py --scripts $(SCRIPTS) --topology t0 --branch $(BUILD_BRANCH))
+            echo "$INSTANCE_NUMBER"
+            echo -n "##vso[task.setvariable variable=INSTANCE_NUMBER]$INSTANCE_NUMBER"
```

Comment:
Does each kvm-test job has this duplicate job? If right, can we make it to a template as well?

Reply:
This is just a sample for T0 PR checker, so we hardcode here. If we add more PR checkers, I will make it as a template.